source/agora/common/BlockStorage.d

```diff
@@ -0,0 +1,292 @@
+/*******************************************************************************
+
+    Defines the memory mapped file of a block.
+
+    The number of files is one.
+    We will expand to use multiple files in the next step.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.BlockStorage;
+
+import agora.common.Block;
+import agora.common.Deserializer;
+import agora.common.Serializer;
+
+import std.algorithm.comparison;
+import std.file;
+import std.format;
+import std.mmfile;
+import std.path;
+
+
+/// Ditto
+public class BlockStorage
+{
+    /// Instance of memory mapped file
+    private MmFile file;
+
+    /// Is already mapped ?
+    private bool is_mapped;
+
+    /// Path of block data
+    private string path;
+
+    /// Index is block height, Value is position of file
+    /// We will change to index file in the next step.
+    private size_t[] block_positions;
+
+    /// Ctor
+    public this (string path)
+    {
+        this.is_mapped = false;
+        this.path = path;
+    }
+
+    public ~this ()
+    {
+        this.unMap();
```

Comment:
You cannot use the GC from a destructor.
Since D has a garbage collector, when a reference is collected by the GC, you have no guarantee that the GC-allocated references it holds are still alive.
If you had this guarantee, it would be impossible to finalize objects with cyclic references.

Reply:
Okay. I'll remove this part.